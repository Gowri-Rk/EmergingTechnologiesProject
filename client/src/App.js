import React, { useState } from 'react';
import AppBar from '@material-ui/core/AppBar';
import { makeStyles } from '@material-ui/core/styles';
import Toolbar from '@material-ui/core/Toolbar'
import TypoGraphy from '@material-ui/core/Typography'
import TextareaAutosize from '@material-ui/core/TextareaAutosize';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
    height: "100%",
    width: "100%",
    padding: 30,
    margin: 5,
  },
  control: {
    padding: theme.spacing(2),
  },
  textArea: {
    width: "100%"
  },
  middle: {
    maxWidth: "fit-content"
  },
  inner: {
    flexDirection: "column"
  },
  dropDown: {
    marginBottom: 20,
  }
}));



export default function App() {
  const classes = useStyles();
  const [content, setContent] = useState("");
  const [summary, setSummary] = useState("");
  const [loading, setLoading] = useState(false);
  const [compression, setCompression] = useState("0.2");
  const [open, setOpen] = React.useState(false);

  const handleChange = (event) => {
    setCompression(event.target.value);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleOpen = () => {
    setOpen(true);
  };

  const fetchSummary = () => {
    setLoading(true);
    fetch('/articles', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        "course": "demo",
        content,
        name: "temp"
      })
    }
    ).then(response => response.json())
      .then(data => {
        fetch(`/articles/${data.id}/summaries`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            ratio: compression,
            name: "temp"
          })
        }
        ).then(response => response.json())
          .then(data => {
            setLoading(false);
            setSummary(data.content);
          })
          .catch((error) => {
            setLoading(false);
            console.error('Error:', error);
          });
      })
      .catch((error) => {
        setLoading(false);
        console.error('Error:', error);
      });
  }
  return (
    <div className="App">
      <AppBar color="primary" position="static">
        <Toolbar>
          <TypoGraphy variant="h4"
            color="inherit"
          >
            BERT Content Summarizer
           </TypoGraphy>
        </Toolbar>
      </AppBar>
      <Grid container
        className={classes.root}
        spacing={2}
        alignItems="center"
        justify="center">
        <Grid key="left" xs={5} item>
          <TextareaAutosize value={content} onChange={(e) => {
            setContent(e.target.value)
          }} className={classes.textArea} aria-label="left textarea" rowsMin={40} placeholder="Paste your content here" />
        </Grid>
        <Grid key="middle" xs={2} className={classes.middle} item>
          <Grid key="inner" className={classes.inner} justify="center" container>
          <FormControl className={classes.formControl}>
            <InputLabel id="demo-simple-select-label">Compression</InputLabel>
            <Select
              labelId="demo-controlled-open-select-label"
              id="demo-controlled-open-select"
              open={open}
              onClose={handleClose}
              onOpen={handleOpen}
              value={compression}
              onChange={handleChange}
              className={classes.dropDown}
            >
              <MenuItem value={0.1}>0.1</MenuItem>
              <MenuItem value={0.2}>0.2</MenuItem>
              <MenuItem value={0.3}>0.3</MenuItem>
            </Select>
          </FormControl>
          <Button variant="contained" color="primary" disabled={loading} onClick={fetchSummary}>
            {loading ? "Generating..." : "> Summarize >"}
          </Button>
          </Grid>
        </Grid>
        <Grid key="right" xs={5} item>
          <TextareaAutosize value={summary} onChange={(e) => {
            setSummary(e.target.value)
          }} className={classes.textArea} aria-label="right textarea" rowsMin={40} placeholder="Summary will be generated here" />
        </Grid>
      </Grid>
    </div>
  );
}
