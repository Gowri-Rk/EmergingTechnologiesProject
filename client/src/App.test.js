import { render, screen } from '@testing-library/react';
import App from './App';
import { shallow } from 'enzyme';

test('App renders properly without any error', () => {
  let wrapped = shallow(<App></App>);
  expect(wrapped).toMatchSnapshot();
});
