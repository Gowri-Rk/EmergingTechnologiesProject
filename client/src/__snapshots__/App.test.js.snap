// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="App"
>
  <WithStyles(ForwardRef(AppBar))
    color="primary"
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        color="inherit"
        variant="h4"
      >
        BERT Content Summarizer
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="makeStyles-root-1"
    container={true}
    justify="center"
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="left"
      xs={5}
    >
      <WithStyles(ForwardRef(TextField))
        className="makeStyles-textArea-3"
        id="url"
        label="URL"
        onChange={[Function]}
        placeholder="Link to url"
        value=""
      />
      <ForwardRef(TextareaAutosize)
        aria-label="left textarea"
        className="makeStyles-textArea-3"
        onChange={[Function]}
        placeholder="Paste your content here"
        rowsMin={40}
        value=""
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-middle-4"
      item={true}
      key="middle"
      xs={2}
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-inner-5"
        container={true}
        justify="center"
        key="inner"
      >
        <WithStyles(ForwardRef(FormControl))>
          <WithStyles(ForwardRef(InputLabel))
            id="demo-simple-select-label"
          >
            Compression
          </WithStyles(ForwardRef(InputLabel))>
          <WithStyles(ForwardRef(Select))
            className="makeStyles-dropDown-6"
            id="demo-controlled-open-select"
            labelId="demo-controlled-open-select-label"
            onChange={[Function]}
            onClose={[Function]}
            onOpen={[Function]}
            open={false}
            value="0.2"
          >
            <WithStyles(ForwardRef(MenuItem))
              value={0.1}
            >
              0.1
            </WithStyles(ForwardRef(MenuItem))>
            <WithStyles(ForwardRef(MenuItem))
              value={0.2}
            >
              0.2
            </WithStyles(ForwardRef(MenuItem))>
            <WithStyles(ForwardRef(MenuItem))
              value={0.3}
            >
              0.3
            </WithStyles(ForwardRef(MenuItem))>
          </WithStyles(ForwardRef(Select))>
        </WithStyles(ForwardRef(FormControl))>
        <WithStyles(ForwardRef(Button))
          color="primary"
          disabled={false}
          onClick={[Function]}
          variant="contained"
        >
          &gt; Summarize &gt;
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="right"
      xs={5}
    >
      <ForwardRef(TextareaAutosize)
        aria-label="right textarea"
        className="makeStyles-textArea-3"
        onChange={[Function]}
        placeholder="Summary will be generated here"
        rowsMin={15}
        value=""
      />
      <img
        alt="wordcloud"
        aria-label="image"
        height={370}
        src=""
        width={550}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
